### PR TITLE
Update trace receiver documentation to align with the most recent updates to open telemetry collector.

### PR DESCRIPTION
**Pull request by gouslu**
This PR is to address outdated trace receiver documentation. Recently I followed this document, and a lot of interfaces and implementations were outdated. Thanks to some knowledge I have about the collector, I was able to find the issues in the document and implement a receiver. As a result I thought I would update the document to reflect these latest changes.

- dev-otelcol changed to otelcol-dev to align with [custom-collector document](https://opentelemetry.io/docs/collector/custom-collector/)'s update.
- updated implementation details, aligning them with the most recent refactoring in the code base of open telemetry collector.
- tested the updated document on myself by following the steps from scratch.

Here is the link to the updated page via netlify: https://deploy-preview-2190--opentelemetry.netlify.app/docs/collector/trace-receiver/

this PR also addresses issue #1802.

### CODE REVIEW

**Comment by Chinwendu20** [comment]
Yeah sure @jpkrohling. On it now

**Comment by Chinwendu20** [comment]
LGTM. Ran the code snippets  on my computer and it works perfectly. Good job @gouslu. Thanks for getting this done

**Review by svrnm — APPROVED** [review]

**Comment by Chinwendu20** [comment]
Does anyone think it would be a good idea to include the link to a GitHub repo that has the complete implementation of the tail trace receiver to the tutorial? Also, it would be great if the version of the collector on the "build custom collector" tutorial is updated to the version you used for this as well. Just for compatibility as there are some significant changes between the two versions.

**Comment by svrnm** [comment]
> Does anyone think it would be a good idea to include the link to a GitHub repo that has the complete implementation of the tail trace receiver to the tutorial? 

Yes, that would be good, could that live in opentelemetry-collector-(contrib) as an example (like we have it for languages) or would we need something additional?

> Also, it would be great if the version of the collector on the "build custom collector" tutorial is updated to the version you used for this as well since it was linked in this tutorial. Just for compatibility as there are some significant changes between the two versions.

👍

**Comment by gouslu** [comment]
Thanks @Chinwendu20 and @svrnm. I don't have the right to merge this PR, can someone do it?

**Comment by cartermp** [comment]
I'd like to get a review from @Aneurysm9 before continuing here (although I agree, the content looks good)

File: content/en/docs/collector/trace-receiver.md
Comment (Aneurysm9) [inline content/en/docs/collector/trace-receiver.md:350]:
```suggestion
The `tailtracer` receiver has to provide a `receiver.Factory` implementation, and
although you will find a `receiver.Factory` interface (you can find its
```

File: content/en/docs/collector/trace-receiver.md
Comment (Aneurysm9) [inline content/en/docs/collector/trace-receiver.md:393]:
```suggestion
- `component.Type`: A component.Type instance representing a unique identifier for
```

File: content/en/docs/collector/trace-receiver.md
Comment (Aneurysm9) [inline content/en/docs/collector/trace-receiver.md:399]:
```suggestion
- `...FactoryOption`: The slice of `receiver.FactoryOption`s that will determine what type of signal your receiver is capable of processing.
```

File: content/en/docs/collector/trace-receiver.md
Comment (Aneurysm9) [inline content/en/docs/collector/trace-receiver.md:407]:
```suggestion
[component.Type](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.69.0/component/config.go#L122),
```

**Review by Aneurysm9 — COMMENTED** [review]
I'm not familiar with this document and there are so many changes that have happened in the collector framework and in this document that I can't have great confidence they're all captured here and correct.  @bogdandrutu has made most of these changes in the collector framework, maybe he can validate this change.

**Comment by gouslu** [comment]
@Aneurysm9 thanks for the review, took your suggested changes, also found some others also that were lacking updates similarly.

**Comment by gouslu** [comment]
@bogdandrutu I think this PR is waiting for your review before it can be merged.

**Comment by cartermp** [comment]
@open-telemetry/collector-approvers can one of y'all give this a look-over? Preferably someone more familiar with the updated trace receiver APIs.

File: content/en/docs/collector/trace-receiver.md
Comment (mx-psi) [inline content/en/docs/collector/trace-receiver.md:97]: I believe this should be
```suggestion
    verbosity: detailed
```

**Review by codeboten — COMMENTED** [review]
Thanks for the update. Overall it looks good just a couple of suggestions.

File: content/en/docs/collector/trace-receiver.md
Comment (chalin) [reply content/en/docs/collector/trace-receiver.md]: Generally speaking, for functions, we want the parentheses: so use `generateTraces()` (as it was originally).

**Comment by chalin** [comment]
As documented in [Submitting a change](https://github.com/open-telemetry/opentelemetry.io/blob/main/CONTRIBUTING.md#submitting-a-change), you'll need to (re-)run the formatter, like this: `npm run check:formatting -- --write`. Then submit the changes, thanks!

File: content/en/docs/collector/trace-receiver.md
Comment (chalin) [inline content/en/docs/collector/trace-receiver.md:503]:
```suggestion
>   a `component.Config` implementation, which in this case is going to be an
```

File: content/en/docs/collector/trace-receiver.md
Comment (chalin) [inline content/en/docs/collector/trace-receiver.md:805]:
```suggestion
Now that you have the `tailtracerReceiver` type you can implement the `Start()`
and `Shutdown()` methods so the receiver type can be compliant with the
```

File: content/en/docs/collector/trace-receiver.md
Comment (chalin) [inline content/en/docs/collector/trace-receiver.md:1831]:
```suggestion
You can now update the `generateTraces()` function and add variables to
```

**Review by chalin — COMMENTED** [review]
A few code-block style nits. See inline suggestions.

File: content/en/docs/collector/trace-receiver.md
Comment (gouslu) [reply content/en/docs/collector/trace-receiver.md]: this conflicts with the other review above, which requested me to completely remove the parenthesis, whereas my original change was from `generateTraces()` to `generateTraces(numberOfTraces int)`. I will leave it as is for now.

**Review by chalin — COMMENTED** [review]
@gouslu - Thanks for the updates. Can we remove line 4, shown next, and the alert from lines 7 to 18?

> collectorVersion: 0.53.0 # Pin to older version until prose is updated

Or is this still pinned to an older version?

**Comment by gouslu** [comment]
> @gouslu - Thanks for the updates. Can we remove line 4, shown next, and the alert from lines 7 to 18?
> 
> > collectorVersion: 0.53.0 # Pin to older version until prose is updated
> 
> Or is this still pinned to an older version?

It should be 0.69.0, I did the changes, but I am not sure why you asked me to remove the collectorVersion. Honestly, I don't think we gain a lot from making it a variable, because there can be more differences between versions than just the version number when it comes to URLs and such. Would you like me to remove collectorVersion and hard code it like it used to be instead? I would like to note that it was not me who did the changes with that collectorVersion thing, it came to my branch after I synced the fork from main, so I am not entirely sure what is the right thing to do here.

**Comment by chalin** [comment]
The latest version of the collector is `0.71.0`. If your changes work with the latest version then the local setting of `collectorVersion` in this page should be removed, as should the `alert`.

**Comment by gouslu** [comment]
> The latest version of the collector is `0.71.0`. If your changes work with the latest version then the local setting of `collectorVersion` in this page should be removed, as should the `alert`.

I verified that it works with 0.69.0 but I did not verify whether it works with 0.71.0. So, I guess I should put the alert back into where it was then? 0.69.0 was the most up-to-date version when I updated this document.

**Review by chalin — COMMENTED** [review]
@svrnm WDYT of the following? I'd vote to drop lines 4, and 7 through 15:

> Screen Shot 2023-02-15 at 19 43 22

**Comment by chalin** [comment]
> I verified that it works with 0.69.0 but I did not verify whether it works with 0.71.0. So, I guess I should put the alert back into where it was then? 0.69.0 was the most up-to-date version when I updated this document.

@gouslu - as mentioned above, I'd vote to drop the alert text and local version number, but I'd like @svrnm et al.'s feedback on this. Thanks!

**Comment by cartermp** [comment]
I'd agree with @chalin's recommendations.

**Comment by chalin** [comment]
@gouslu - can you make the updates described in https://github.com/open-telemetry/opentelemetry.io/pull/2190#pullrequestreview-1300590004?

**Comment by gouslu** [comment]
@chalin done, can you please verify if all looks good?

**Review by chalin — APPROVED** [review]
LGTM in that the alert and version-pin has been removed. Thanks!

**Comment by chalin** [comment]
You'll need to rebase, resolve conflicts and push.

**Comment by chalin** [comment]
When you rebase, you'll probably want to just ignore incoming changes (from #2315, which reformat the file) and just override with your changes. Sorry about that.

**Comment by gouslu** [comment]
@chalin after the rebase checks started failing, and I cannot figure out why?

File: content/en/docs/collector/trace-receiver.md
Comment (chalin) [inline content/en/docs/collector/trace-receiver.md:113]:
```suggestion
      level: debug
```

File: content/en/docs/collector/trace-receiver.md
Comment (chalin) [inline content/en/docs/collector/trace-receiver.md:236]: Do not hard code collector versions, instead use the version variable like this (make this change everywhere in the document):

```suggestion
[ConfigValidator](https://github.com/open-telemetry/opentelemetry-collector/blob/{{% param collectorVersion %}}/component/config.go#L50)
```

**Review by chalin — CHANGES_REQUESTED** [review]
See inline comments.

**Comment by chalin** [comment]
> @chalin after the rebase checks started failing, and I cannot figure out why?

Looking at the Git commit graph, it seems that you did a _merge_ instead of a rebase:

> Screen Shot 2023-02-24 at 05 02 31

Let me see if I can fix that for you.

**Comment by chalin** [comment]
@gouslu - merge undone and rebase completed. The build is passing now. I also committed some fixes to the code-block languages. So you'll need to pull changes to your local machine before making any further changes.

File: content/en/docs/collector/trace-receiver.md
Comment (chalin) [reply content/en/docs/collector/trace-receiver.md:236]: Actually, since I'm in the file now, let me make those changes.

Comment (chalin) [reply content/en/docs/collector/trace-receiver.md]: Done

**Review by chalin — APPROVED** [review]
LGTM (the opening paragraph could benefit from some copyediting, but this will do for now).

**Comment by gouslu** [comment]
@chalin thanks for all the help

**Comment by gouslu** [comment]
I did a cleanup and updated the PR. If all still looks good, can someone please merge it?

Thanks a lot!

**Comment by chalin** [comment]
@gouslu - I don't know how your are doing your rebases, but it is breaking the PR and we're losing changes, sadly: for example, the `collectorVersion` is now back in the front matter.

I suggest that we leave the PR as is. Let's get this approved and merged, and I'll make a followup PR for the remaining minor changes. /cc @svrnm

**Review by cartermp — APPROVED** [review]

**Pull request merged**
